src/com/connectsdk/device/ConnectableDevice.java

```diff
@@ -129,6 +129,25 @@ public ConnectableDevice(JSONObject json) {
         setLastSeenOnWifi(json.optString(KEY_LAST_SEEN, null));
         setLastConnected(json.optLong(KEY_LAST_CONNECTED, 0));
         setLastDetection(json.optLong(KEY_LAST_DETECTED, 0));
+
+        //Deserialize the associated services as well
+        try {
+            JSONObject jsonServices = json.getJSONObject(KEY_SERVICES);
```

Comment:
You should probably use 'optJSONObject here, just like the piece of code above this.